openwisp_firmware_upgrader/tests/test_models.py

```diff
@@ -310,6 +310,75 @@ def test_upgrade_operation_retention_on_image_delete(self):
         FirmwareImage.objects.get(pk=device_fw.image.pk).delete()
         self.assertEqual(UpgradeOperation.objects.get(pk=uo.pk).image, None)
 
+    def test_delete_firmware_image_file(self):
+        file_storage_backend = FirmwareImage.file.field.storage
+
+        with self.subTest("Test deleting object deletes file"):
+            image = self._create_firmware_image()
+            file_name = image.file.name
+            image.delete()
+            self.assertEqual(file_storage_backend.exists(file_name), False)
+
+        with self.subTest("Test deleting object with a deleted file"):
+            image = self._create_firmware_image()
+            file_name = image.file.name
+            # Delete the file from the storage backend before
+            # deleting the object
+            file_storage_backend.delete(file_name)
+            self.assertNotEqual(image.file, None)
+            image.delete()
+
+    def test_delete_firmware_files_on_build_delete(self):
+        """Test that firmware files are deleted when a build is deleted"""
+        file_storage_backend = FirmwareImage.file.field.storage
+        build = self._create_build()
+        image = self._create_firmware_image(build=build)
+        file_name = image.file.name
+
+        # Delete the build
+        build.delete()
+
+        # Execute the Celery task synchronously
+        delete_firmware_files([file_name])
```

Comment:
Why is calling this explicitly needed? Isn't this supposed to be called automatically due to `CELERY_TASK_ALWAYS_EAGER = True`?